### PR TITLE
Add some information in the documentation for beginners.

### LINKED ISSUE #238: Add more precise directions on how to code/test in the Contributing documentation
I am willing to contribute to Watson. I followed the instructions currently in the http://tailordev.github.io/Watson/contributing/hack page, everything is OK (forking, virtualenv, branch creation, tox).
However, I am missing some directions on how to work at the `Now you can make your changes locally.` step. 

I have a watson installed on my machine, and the development version. What files from my development repo should I edit? I guess they are in the `watson` subfolder. I see a  `build/lib/watson` subfolder, I guess I have to use `make` or something so that files from `watson/` are moved to `build/lib/watson`: which command(s) do(es) that?

Also, once I will have made some changes (and *built* them?), how do I test the new watson behavior/command?  

For instance, I get this error:
```
$ python watson/watson.py
Traceback (most recent call last):
  File "watson/watson.py", line 18, in <module>
    from .config import ConfigParser
ModuleNotFoundError: No module named '__main__.config'; '__main__' is not a package
```
I would also like to use some frames from my production watson, to test some features, but would not like to mess them up. How may I copy some of them so that they are used by the development version?

### PR DESCRIPTION
**Pull request by raphaelfournier**
Questions raised from the previous version and answered by the new paragraphs:
- Which files should one edit after setting up the environment?
- How to copy some real frames from a working watson install towards the
development one?
- How to set up the environment variable WATSON_DIR?
- How to avoid the incompatibility between mkdocs 0.14 and Python 3.7?

Most information was given by @jmaupetit in https://github.com/TailorDev/Watson/issues/238.

Fix #238

### CODE REVIEW

File: docs/contributing/hack.md [hunk 1]
```diff
@@ -30,6 +69,12 @@ Ready to contribute? Here's how to set up *Watson* for local development.
 6. If you have added a new command or updated/fixed docstrings, please update the documentation:
 
         $ make docs
+   
+     Beware that mkdocs 0.14 is not compatible with Python 3.7: you should either
+     use a virtualenv with Python 3.6 or update `mkdocs` to a more recent version
+     (1.0.4 at the time of writing, which also requires running `pip install
+     --upgrade mkdocs-bootstrap mkdocs-bootswatch` to get the `flatly` theme
+     used by our docs).
```
Comment (k4nar) [inline docs/contributing/hack.md]: I think we should rather update our requirements so that we don't install `mkdocs` <= 0.14.

Comment (jmaupetit) [reply docs/contributing/hack.md]: Yes, but it should be handled in a separated PR @raphaelfournier :wink:

**Comment by davidag** [comment]
It looks like this PR might be useful for beginners, are you planning to merge it? is there anything I can do to improve it? Maybe it could benefit from some simplification.

**Comment by jmaupetit** [comment]
@raphaelfournier can you rebase this work on the `master` branch and avoid merge commit?

File: docs/contributing/hack.md [hunk 2]
```diff
@@ -15,13 +15,52 @@ Ready to contribute? Here's how to set up *Watson* for local development.
...
+      
+        It may be a good idea to modify the `bin/activate` file from your virtualenv
+        (either `venv/bin/activate` or `$HOME/.virtualenvs/watson/bin/activate`),
+        and add the following lines (just before the `PATH` settings, for instance):
+        
+            WATSON_DIR=$PWD/data
+            export WATSON_DIR
+        
+        This will be set each time you start the virtualenv and you will not have to
+        re-set the variable each time.
```
Comment (jmaupetit) [inline docs/contributing/hack.md]: Depending on #273 status, you may need to update this section accordingly.

File: docs/contributing/hack.md [hunk 3]
```diff
@@ -15,13 +15,52 @@ Ready to contribute? Here's how to set up *Watson* for local development.
...
 
 4.  Create a branch for local development:
 
         $ git checkout -b name-of-your-bugfix-or-feature
 
     Now you can make your changes locally.
-
+    
+    The files you need to edit to change watson's behavior are located in the
```
Comment (jmaupetit) [inline docs/contributing/hack.md]: Watson stands for the project, and `watson` for the command line tool:

```suggestion
    The files you need to edit to change Watson's behavior are located in the
```

Comment (raphaelfournier) [reply docs/contributing/hack.md]: Sorry guys, I haven't been working on developing Watson for a while, due to other professional activities. I will resume and close this PR at the beginning of the next week.

Comment (jmaupetit) [reply docs/contributing/hack.md]: Don't worry, we are all volunteers here! :muscle:

Comment (davidag) [reply docs/contributing/hack.md]: @jmaupetit would it be ok if I continue working on these in a different PR?

Comment (jmaupetit) [reply docs/contributing/hack.md]: I think you should ask @raphaelfournier first :wink:

Comment (davidag) [reply docs/contributing/hack.md]: Of course! @raphaelfournier what do you think? can I help you close this PR somehow?

Comment (jmaupetit) [reply docs/contributing/hack.md]: No worries!

File: docs/contributing/hack.md (hunk 1 above)
Comment (davidag) [reply docs/contributing/hack.md]: Added issue #280 to handle this.

**Comment by davidag** [comment]
@jmaupetit I believe that after merging #304 and #279 this pull request can be safely closed, what do you think? cc/ @raphaelfournier

**Comment by jmaupetit** [comment]
Probably. I think we should ask @raphaelfournier its tougths about it. Raphaël, do you think recent changes from @davidag fulfill your expectations about Watson's documentation for beginners?

**Comment by raphaelfournier** [comment]
Hello @jmaupetit and @davidag, thank you very much for your work on this! Yes, I think it is perfectly okay to close this PR. 
My paragraph on the "local changes" was very "I will hold your hand", and most contributors don't need it, the important bit is preserved (the files to edit are in the watson subfolder).
Thank you for your work!

**Pull request closed without merging**